groups/validate.go

```diff
@@ -104,196 +134,232 @@ func TestMergedGroupsConfig(t *testing.T) {
+				log.Printf("group '%s': must have no owners, only members", g.Name)
+				errs++
 			}
 
 			// treat files here as source of truth for membership
 			reconcileMembers, ok := g.Settings["ReconcileMembers"]
 			if !ok || reconcileMembers != "true" {
-				t.Errorf("group '%s': must have settings.ReconcileMembers = true", g.Name)
+				log.Printf("group '%s': must have settings.ReconcileMembers = true", g.Name)
+				errs++
 			}
 		}
 	}
+	if errs > 0 {
+		os.Exit(1)
+	}
 }
 
 // Enforce conventions for groups used by GKE Group-based RBAC
 // - there must be a gke-security-groups@ group
 // - its members must be k8s-infra-rbac-*@ groups (and vice-versa)
 // - all groups involved must have settings.WhoCanViewMembership = ALL_MEMBERS_CAN_VIEW
-func TestK8sInfraRBACGroupConventions(t *testing.T) {
+func VerifyK8sInfraRBACGroupConventions(config Config) {
+	primaryDomain := config.Domains[0]
 	rbacEmails := make(map[string]bool)
-	for _, g := range cfg.Groups {
+	errs := 0
+	for _, g := range groupsConfig.Groups {
 		if strings.HasPrefix(g.EmailId, "k8s-infra-rbac") {
 			rbacEmails[g.EmailId] = false
 			// this is necessary for group-based rbac to work
 			whoCanViewMembership, ok := g.Settings["WhoCanViewMembership"]
 			if !ok || whoCanViewMembership != "ALL_MEMBERS_CAN_VIEW" {
-				t.Errorf("group '%s': must have settings.WhoCanViewMembership = ALL_MEMBERS_CAN_VIEW", g.Name)
+				log.Printf("group '%s': must have settings.WhoCanViewMembership = ALL_MEMBERS_CAN_VIEW", g.Name)
+				errs++
 			}
 		}
 	}
 	foundGKEGroup := false
-	for _, g := range cfg.Groups {
-		if g.EmailId == "gke-security-groups@kubernetes.io" {
+	for _, g := range groupsConfig.Groups {
+		if g.EmailId == "gke-security-groups@"+primaryDomain {
 			foundGKEGroup = true
 			// this is necessary for group-based rbac to work
 			whoCanViewMembership, ok := g.Settings["WhoCanViewMembership"]
 			if !ok || whoCanViewMembership != "ALL_MEMBERS_CAN_VIEW" {
-				t.Errorf("group '%s': must have settings.WhoCanViewMembership = ALL_MEMBERS_CAN_VIEW", g.Name)
+				log.Printf("group '%s': must have settings.WhoCanViewMembership = ALL_MEMBERS_CAN_VIEW", g.Name)
+				errs++
 			}
 			for _, email := range g.Members {
 				if _, ok := rbacEmails[email]; !ok {
-					t.Errorf("group '%s': invalid member '%s', must be a k8s-infra-rbac-*@kubernetes.io group", g.Name, email)
+					log.Printf("group '%s': invalid member '%s', must be a k8s-infra-rbac-*@"+primaryDomain+"group", g.Name, email)
+					errs++
 				} else {
 					rbacEmails[email] = true
 				}
 			}
 		}
 	}
 	if !foundGKEGroup {
-		t.Errorf("group '%s' is missing", "gke-security-groups@kubernetes.io")
+		log.Printf("group '%s' is missing", "gke-security-groups@"+primaryDomain)
+		errs++
 	}
 	for email, found := range rbacEmails {
 		if !found {
-			t.Errorf("group '%s': must be a member of gke-security-groups@kubernetes.io", email)
+			log.Printf("group '%s': must be a member of gke-security-groups@"+primaryDomain, email)
```

Comment:
A 👎  on combining `log.Printf` and string `+` -- but I'd switch these back to `@kubernetes.io` anyway.